src/jit/rationalize.cpp

```diff
@@ -157,6 +157,13 @@ void Rationalizer::RewriteNodeAsCall(GenTree**             use,
     // Create the call node
     GenTreeCall* call = comp->gtNewCallNode(CT_USER_FUNC, callHnd, tree->gtType, args);
     call              = comp->fgMorphArgs(call);
+    // Determine if this call has changed any codegen requirements.
+    comp->fgCheckArgCnt();
+    if ((call->gtFlags & GTF_EXCEPT) != 0)
+    {
+        comp->codeGen->setFramePointerRequiredEH(true);
+    }
```

Comment:
We call setFramePointerRequiredEH() only when EH is the reason hat we require EBP to be reserved.

For this case it is a different reason that EBP is reserved.  (Presumably so we can create a EBP frame chain for ETW)

So you should add a new method setFramePointerRequiredNonLeaf() that record the reason